version: '3.4'

services:
  traefik:
    image: traefik:v2.11
    container_name: traefik
    command:
      - --configFile=/etc/traefik/traefik.yml
    ports:
      - "80:80"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./traefik/traefik.yml:/etc/traefik/traefik.yml:ro
    networks:
      - web

  django:
    container_name: django
    build:
      context: ./django
      dockerfile: ./Dockerfile
    stdin_open: true   
    tty: true   
    volumes:
     - ./output:/output
     - ./django:/home/app/webapp  
    environment:
      - DOCKER_RUNNING=true
      - JIRA_SERVER_URL=https://yourdomain.atlassian.net
      - JIRA_API_TOKEN=${JIRA_API_TOKEN}
      - JIRA_USER_EMAIL=${JIRA_USER_EMAIL}
      - S1_BASE_URL=${S1_BASE_URL}
      - S1_API_TOKEN=${S1_API_TOKEN}
      - S1_CLIENT_ID=${S1_CLIENT_ID}
      - S1_CLIENT_SECRET=${S1_CLIENT_SECRET}
      - S1_SITE_ID=${S1_SITE_ID}
    env_file:
      - ./config/config.ini 
    depends_on: 
      db:
        condition: service_healthy
    healthcheck:
      test: wget --no-verbose -O /dev/null --tries=1 http://localhost:8000/api/health-check || exit 1
      interval: 10s
      timeout: 5s 
    networks:
      - web
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`django.localhost`)"

  mesop-app:
    container_name: mesop
    build:
      context: ./mesop
      dockerfile: ./Dockerfile
    volumes:
     - ./mesop/src:/srv/mesop-app/src
    environment:
      - PYTHONUNBUFFERED=1
      - DOCKER_RUNNING=true
    depends_on: 
      django:
        condition: service_healthy
    env_file:
      - ./config/config.ini 
    networks:
      - web
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.mesop.rule=Host(`mesop.localhost`)"

  db:
    hostname: db
    image: pgvector/pgvector:pg15
    ports:
     - 5432:5432
    environment:
      POSTGRES_DB: vectordb
      POSTGRES_USER: testuser
      POSTGRES_PASSWORD: testpwd
      PGUSER: testuser
      POSTGRES_HOST_AUTH_METHOD: trust
      PGDATA: /var/lib/postgresql/data/
    volumes:
      - ./pgvector/output/database:/var/lib/postgresql/data
      - ./pgvector/init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d vectordb -U testuser"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - web

networks:
  web:
    external: false